org_fedora_hello_world/gui/spokes/hello_world.py

```diff
@@ -112,28 +116,32 @@ def refresh(self):
         self.data.
 
         :see: pyanaconda.ui.common.UIObject.refresh
-
         """
-
-        self._entry.set_text(self.data.addons.org_fedora_hello_world.text)
+        lines = self._hello_world_module.Lines
+        self._entry.get_buffer().set_text("".join(lines))
+        reverse = self._hello_world_module.Reverse
+        self._reverse.set_active(reverse)
 
     def apply(self):
         """
         The apply method that is called when the spoke is left. It should
-        update the contents of self.data with values set in the GUI elements.
-
+        update the service with values set in the GUI elements.
         """
+        buf = self._entry.get_buffer()
+        text = buf.get_text(buf.get_start_iter(),
+                            buf.get_end_iter(),
+                            True)
+        lines = text.splitlines(True)
+        self._hello_world_module.SetLines(lines)
 
-        self.data.addons.org_fedora_hello_world.text = self._entry.get_text()
```

Comment:
Why `self._entry.get_text()` is not sufficient?

Reply:
Previously, `_entry` was a single line edit, but now it is a multiline memo (wrong terminology probably). These have a wildly different interface, sadly.

Maybe I should rename the field to make it clearer it's now something different? Not sure who beside a reviewer it helps, though...